.github/copilot-instructions.md

```diff
@@ -1,92 +1,123 @@
+- **Test with logging**: `RAILS_ENABLE_TEST_LOG=true bundle exec rspec path/to/spec.rb` (logs to `log/test.log`)
+- **Lint**: `bundle exec rubocop` (handled by CI, don't suggest fixes)
+- **Server**: `foreman start -m all=1`
+- **Console**: `bundle exec rails console`
+- **DB**: `make db` (setup + migrate)
+
+### Quick Reference
+- **Config files**: `modules/[name]/config/` or main `config/`
+- **Serializers**: End controllers with `render json: object, serializer: SomeSerializer`
+- **Auth**: Most endpoints need `before_action :authenticate_user!`
+- **Jobs**: Use `perform_async` for background work, `perform_in` for delayed
+- **Flipper in tests**: Never use `Flipper.enable/disable` - always stub with `allow(Flipper).to receive(:enabled?).with(:feature).and_return(true)`
+
+### Common Patterns
+- Controllers in `modules/[name]/app/controllers` or `app/controllers`
+- Background jobs in `app/sidekiq/` - use for operations >2 seconds
+- External service clients in `lib/` with Faraday configuration
+- Feature flags via Flipper for gradual rollouts and A/B testing
+- Strong parameters required - never use `params` directly
+- Error responses use envelope: `{ error: { code, message } }`
+- Service objects return `{ data: result, error: nil }` pattern
+
+### Key Dependencies
+- PostGIS required for database
+- Sidekiq Enterprise (may need license)
+- VCR cassettes for external service tests
+- Settings: `config/settings.yml` (alphabetical order required)
+
+### VA Service Integration
+- **BGS**: Benefits data, often slow/unreliable
+- **MVI**: Veteran identity, use ICN for lookups
+- **Lighthouse**: Modern REST APIs for claims, health records, veteran verification
+
+## For PR Reviews - Human Judgment Issues
+
+### ⚠️ NO DUPLICATE COMMENTS - Consolidate Similar Issues
+
+### Security & Privacy Concerns
+- **PII in logs**: Check for email, SSN, medical data in log statements
+- **Hardcoded secrets**: API keys, tokens in source code
+- **Missing authentication**: Controllers handling sensitive data without auth checks
+- **Mass assignment**: Direct use of params hash without strong parameters
+
+### Business Logic Issues
+- **Non-idempotent operations**: Creates without duplicate protection
+- **Blocking operations in controllers**: sleep(), File.read, document processing, operations >2 seconds
+- **Wrong error response format**: Not using VA.gov standard error envelope
+- **Service method contracts**: Returning `{ success: true }` instead of data/error pattern
+
+### VA-Specific Patterns
+- **New logging without Flipper**: Logs not wrapped with feature flags
+- **External service calls**: Missing error handling, timeouts, retries, or rescue blocks
+- **Background job candidates**: File.read operations, PDF/document processing, bulk database updates, .deliver_now emails
+- **Wrong identifier usage**: Using User ID instead of ICN for MVI/BGS lookups
+- **Form handling**: Complex forms not using form objects for serialization
```

Comment:
@stiehlrod Now that these are all re-worded is **VA-Specific Patterns** the right header for this section?